### PR TITLE
[BugFix] Fix partial update fail in replicated storage

### LINKED ISSUE #15661: Partial update fail when load replicated storage table
### Steps to reproduce the behavior (Required)
1. 
```
CREATE TABLE `t` (
  `id` bigint(20) NOT NULL COMMENT "",
  `name` bigint(20) NOT NULL COMMENT "",
  `job1` bigint(20) NULL COMMENT "",
  `job2` bigint(20) NULL COMMENT ""
) ENGINE=OLAP
PRIMARY KEY(`id`, `name`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`id`) BUCKETS 10
PROPERTIES (
"replication_num" = "3",
"in_memory" = "false",
"storage_format" = "DEFAULT",
"enable_persistent_index" = "false",
"replicated_storage" = "true",
"compression" = "LZ4"
);
```
2. INSERT INTO '....'
```
insert into t values (0,0,-1,-1);
```
3. 
```
curl --location-trusted -u root: -H "column_separator:," -H "label:l1" -H "columns:id,name,job2" -H "partial_update:true" -H "max_filter_ratio:0.2" -T data.csv -XPUT http://0.0.0.0:8040/api/load_benchmark/t/_stream_load

cat data.csv
0,0,99
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #15661

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by Linkerist — APPROVED** [review]

**Review by rickif — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/storage/rowset/rowset_writer.cpp [hunk 1]
```diff
@@ -226,6 +228,12 @@ Status RowsetWriter::flush_segment(const SegmentPB& segment_pb, butil::IOBuf& da
         }
         RETURN_IF_ERROR(wfile->close());
 
+        if (_context.tablet_schema->keys_type() == KeysType::PRIMARY_KEYS && _context.partial_update_tablet_schema) {
```
Comment (chaoyli) [inline be/src/storage/rowset/rowset_writer.cpp:231]: 1. The issue should describe the error phenomenon in the expected behavior and current behavior.
2. Better to add a concise message about the problem and solution in the commit message.

File: be/src/storage/rowset/rowset_writer.cpp [hunk 2]
```diff
@@ -161,7 +161,9 @@ StatusOr<RowsetSharedPtr> RowsetWriter::build() {
         if (_context.partial_update_tablet_schema && _flush_chunk_state != FlushChunkState::DELETE) {
             DCHECK(_context.referenced_column_ids.size() == _context.partial_update_tablet_schema->columns().size());
             RETURN_IF(_num_segment != _rowset_txn_meta_pb->partial_rowset_footers().size(),
-                      Status::InternalError("segment number not equal to partial_rowset_footers size"));
+                      Status::InternalError(fmt::format("segment number {} not equal to partial_rowset_footers size {}",
```
Comment (chaoyli) [inline be/src/storage/rowset/rowset_writer.cpp:164]: better add a tablet_id in the error info, nice to trace

**Review by gengjun-git — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by meegoo** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**